apps/emqx/src/emqx_schema.erl

```diff
@@ -3385,3 +3406,27 @@ ensure_default_listener(Map, ListenerType) ->
 
 cert_file(_File, client) -> undefined;
 cert_file(File, server) -> iolist_to_binary(filename:join(["${EMQX_ETC_DIR}", "certs", File])).
+
+-define(EQUAL(V, Default), round(V * 100) =:= round(Default * 100)).
+
+mqtt_converter(#{<<"keepalive_multiplier">> := Multi} = Mqtt, _Opts) ->
+    case ?EQUAL(Multi, ?DEFAULT_MULTIPLIER) of
+        %% change multiplier
+        false ->
+            Mqtt;
+        %% default multiplier, fallback to backoff
+        true ->
+            Backoff = maps:get(<<"keepalive_backoff">>, Mqtt, ?DEFAULT_BACKOFF),
+            Mqtt#{<<"keepalive_multiplier">> => Backoff * 2}
+    end;
+mqtt_converter(#{<<"keepalive_backoff">> := Backoff} = Mqtt, _Opts) ->
+    case ?EQUAL(Backoff, ?DEFAULT_BACKOFF) of
+        %% change backoff and multiplier is default
+        false ->
+            Mqtt#{<<"keepalive_multiplier">> => Backoff * 2};
+        %% backoff and multiplier both are default.
+        true ->
+            Mqtt
+    end;
```

Comment:
```suggestion
    Backoff = ?EQUAL(Backoff, ?DEFAULT_BACKOFF),
    Mqtt#{<<"keepalive_multiplier">> => Backoff * 2};
```

Reply:
when someone upgrade from old version, there is `backoff = 1.5`
after upgrade, he can't see backoff, then change multiplier from default to 4,
IMO, we should apply multiplier = 4, not 1.5*2 ?